### PR TITLE
[MaterialDatePicker] Crash Fix - Ensure calendar stays within valid constraints

### PR DESCRIPTION
**Pull request by zkovar**
### Thanks for starting a pull request on Material Components!

Closes: #980 

#### Don't forget:

- [x] Identify the component the PR relates to in brackets in the title.
  `[Buttons] Updated documentation`
- [x] Link to GitHub issues it solves. `closes #1234`
- [x] Sign the CLA bot. You can do this once the pull request is opened.

[Contributing](https://github.com/material-components/material-components-android/blob/master/docs/contributing.md)
has more information and tips for a great pull request.

### CODE REVIEW

**Comment by zkovar** [comment]
@pekingme Any chance I can get a review on this? We're seeing some user crashes due to this issue.

File: lib/java/com/google/android/material/datepicker/MaterialCalendar.java [hunk 1]
```diff
@@ -298,6 +298,19 @@ CalendarConstraints getCalendarConstraints() {
   void setCurrentMonth(Month moveTo) {
     MonthsPagerAdapter adapter = (MonthsPagerAdapter) recyclerView.getAdapter();
     int moveToPosition = adapter.getPosition(moveTo);
+
+    // If target position is outside of the constraint bounds, adjust target to outer bounds
+    if (calendarConstraints != null) {
+      if (moveToPosition > calendarConstraints.getMonthSpan()) {
```
Comment (klblk) [inline lib/java/com/google/android/material/datepicker/MaterialCalendar.java]: Probably need change to:
```
moveToPosition >= calendarConstraints.getMonthSpan()
```

Comment (klblk) [reply lib/java/com/google/android/material/datepicker/MaterialCalendar.java]: and below
```
int offset = moveToPosition - calendarConstraints.getMonthSpan() + 1;
```

Comment (zkovar) [reply lib/java/com/google/android/material/datepicker/MaterialCalendar.java]: @klblk I do not believe your suggestions are correct. `moveToPosition` is the position within the recyclerView. If `movePosition == calendarConstraints.getMonthSpan()` this indicates we will just be scrolling to the end of it and there is no need to offset the target position. This holds true for the minimum scenario as well and was verified in my personal testing. Is there a counter-case that you encountered that makes you think my solution is "off-by-one" ?

Comment (klblk) [reply lib/java/com/google/android/material/datepicker/MaterialCalendar.java]: @zkovar I debugged option "one year forward". 
February 2020 -> position = 0
February 2021 -> position = 12
calendarConstraints.getMonthSpan() = 13

if I select March 2020 and next switch year to 2021 -> movePosition = 13 and `moveToPosition > calendarConstraints.getMonthSpan()` false

but position 13 -> March 2021 and not in the range

**Comment by ymarian** [comment]
Hi @zkovar , thanks for the pull request I ended up doing a bit of a different fix which I think makes more sense semantics wise. Please take a look!

**Comment by zkovar** [comment]
> Hi @zkovar , thanks for the pull request I ended up doing a bit of a different fix which I think makes more sense semantics wise. Please take a look!

Looks good to me, thanks for tackling this 👍

**Pull request closed without merging**